cli/azd/pkg/azure/tags.go

```diff
@@ -7,6 +7,9 @@ const (
 	// TagKeyAzdEnvName is the name of the key in the tags map of a resource
 	// used to store the azd environment a resource is associated with.
 	TagKeyAzdEnvName = "azd-env-name"
+	// TagKeyAzdDeploymentStateParamHashName is the name of the key in the tags map of a deployment
+	// used to store the parameters hash.
+	TagKeyAzdDeploymentStateParamHashName = "azd-deploy-param-hash-name"
```

Comment:
Perhaps this should be `azd-provision-param-hash`. (since it's used by `provision` and not `deploy` and it's a value, not a name?)